### PR TITLE
Expand evaluation of global built-ins in `@babel/traverse`

### PR DESCRIPTION
**Pull request by lorenzoferre**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            |
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      | Yes
| Tests Added + Pass?      |  Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Evaluation of global built-ins includes `btoa` and `atob`. Before this pull request `path.evaluate()` of `isFinite`, `isNaN`, `parseFloat`, `parseInt`, `encodeURI`, `decodeURI`, `encodeURIComponent` and `decodeURIComponent` were possibile only with keywords `String` or `Number` before the method name (`Number.parseInt("1");`).
An example of this new feature:
```js
const oldTraverse = require("@babel/traverse").default;
const newTraverse = require("../babel/packages/babel-traverse/lib").default;
const babel = require("@babel/core");

const code = "parseInt('1');";

var ast = babel.parse(code);

oldTraverse(ast, {
  CallExpression(path) {
    let evaluated = path.evaluate();
    console.log(`confident old traverse: ${evaluated.confident}`);
  }
});

newTraverse(ast, {
  CallExpression(path) {
    let evaluated = path.evaluate();
    console.log(`confident new traverse: ${evaluated.confident}`);
    console.log(`value new traverse: ${evaluated.value}`);
  }
});
```
This code produces:
```js
confident old traverse: false
confident new traverse: true
value new traverse: 1
```

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15797"></a>

### CODE REVIEW

**Comment by lorenzoferre** [comment]
There is no backwards compatibility with older versions of node due to operators like `?.`, `||=` and so on inside the `packages/babel-traverse/lib` files.

**Comment by JLHwung** [comment]
> operators like ?., ||= and so on inside the packages/babel-traverse/lib files.

If you mean you have spotted them on your local environment, we compile `src` to `lib` targeting the `current` node version, i.e. the node version you are using locally, which probably supports these operators. On CI, we will compile `src` targeting Node.js 6.9, the minimum Node version Babel 7 (still) supports.

**Comment by lorenzoferre** [comment]
@JLHwung this is my first time making a pull request, so do I need to do anything else before someone review my code?

**Comment by JLHwung** [comment]
> btoa and atob functions are available from version 16 onwards.

You are right. In that case we should postpone the support otherwise `path.evaluate()` will return different results on different node versions.

**Review by JLHwung — APPROVED** [review]

**Comment by liuxingbaoyu** [comment]
Can you add a test that might throw an exception? For example `atob("a")`

**Comment by lorenzoferre** [comment]
> Can you add a test that might throw an exception? For example `atob("a")`

`path.evaluate()` of btoa and atob produces undefined if the node version is lower than 16, so I updated the tests with this undefined check.
If, on the other hand, one tries from the node console to use btoa or atob, it throws an Uncaught ReferenceError exception, but I wouldn't know how to test throwing the exception.

File: packages/babel-traverse/src/path/evaluation.ts
Comment (JLHwung) [inline packages/babel-traverse/src/path/evaluation.ts:20]:
```suggestion
  process.env.BABEL_8_BREAKING ? "btoa" : null,
  process.env.BABEL_8_BREAKING ? "atob" : null,
```
The `btoa` and `atob` support should be removed for Babel 7, because given the exact config, running Babel on Node.js 16 should produce output as same as the one generated on Node.js 14.

In Babel 8 we will drop Node 14 support, so we can add `btoa` and `atob` support then.

**Review by JLHwung — CHANGES_REQUESTED** [review]

File: packages/babel-traverse/src/path/evaluation.ts
Comment (lorenzoferre) [reply packages/babel-traverse/src/path/evaluation.ts]: All right, I didn't know it. Now I should have updated everything.

**Comment by lorenzoferre** [comment]
@JLHwung I also have updated the btoa and atob tests.
I think they are cleaner than the old version.

File: packages/babel-traverse/test/evaluation.js
Comment (JLHwung) [inline packages/babel-traverse/test/evaluation.js:199]:
```suggestion
```
Since they are features in Babel 8 now.

**Review by JLHwung — APPROVED** [review]

File: packages/babel-traverse/test/evaluation.js
Comment (lorenzoferre) [reply packages/babel-traverse/test/evaluation.js]: Thank you very much, I forgot to delete that comment

**Comment by lorenzoferre** [comment]
@JLHwung Sorry for my many questions but now this pull request could be merged?
Because I don't know how the flow of these kind of operation works.

**Comment by nicolo-ribaudo** [comment]
Actually, since this is a new feature it should be landed in a minor version (7.23.0). You will have to wait a bit before that this PR is merged (probably ~1 month? maybe less, you can track progress by seeing which PRs in https://github.com/babel/babel/milestone/57 have the two approvals), but there is nothing else that you need to do :)

**Comment by lorenzoferre** [comment]
All right, thank you very very much @JLHwung and @nicolo-ribaudo for everything. As a I said before this is my first pull request so I didn't know if I needed do anything else and I'm sorry for my many questions.
I'm very glad to contribute to a project like this.
I made this pull request because I'm using babel in my thesis project about automating the javascript deobfuscation process via abstract syntaxt trees. In this scenario it's very common to find operations like parseInt, parseFloat etc. in obfuscated code.

**Comment by nicolo-ribaudo** [comment]
That's awesome :) If you are graduating in September and you are mentioning this PR in your thesis, we can make sure to release it by then.

**Comment by lorenzoferre** [comment]
> That's awesome :) If you are graduating in September and you are mentioning this PR in your thesis, we can make sure to release it by then.

Yeah, I should graduate on 28 or 29 September or 2 October.
Certainly, if there is the possibility of mentioning this pull request I am more than happy and I do it very willingly.

**Comment by nicolo-ribaudo** [comment]
We'll release today, sorry for this being so last-minute :)

**Comment by lorenzoferre** [comment]
> We'll release today, sorry for this being so last-minute :)

That's fantastic. 
You don't sweat it, I'm on time. 
I really appreciate that you released this version before my graduation.

**Pull request merged**
